### ISSUE TITLE
How to avoid an external call to refresh  credentials when using the bigtable emulator locally for tests?

### ISSUE THREAD

**Issue by ghost** [GoogleCloudPlatform/cloud-bigtable-client#1607]
We're using the [bigtable emulator](https://cloud.google.com/bigtable/docs/emulator) in our tests. We establish a connection to the running emulator like this:
```
// our own code

  public void connect() throws IOException {
    if (PortChecker.isPortActive(8086)) {
      this.bigtableConnection = this.getEmulatorConnection();
    } else {
      this.bigtableConnection = this.getCloudConnection();
    }
  }

  private BigtableConnection getEmulatorConnection() throws IOException {
    Configuration conf = BigtableConfiguration.configure("unit-test-project", "unit-test-instance");
    conf.set("google.bigtable.instance.admin.endpoint.host", "localhost");
    conf.set("google.bigtable.admin.endpoint.host", "localhost");
    conf.set("google.bigtable.endpoint.host", "localhost");
    conf.set("google.bigtable.endpoint.port", "8086");
    conf.set("google.bigtable.use.plaintext.negotiation", "true");
    conf.set("google.bigtable.grpc.read.partial.row.timeout.ms", "5000");
    BigtableConnection connection = new BigtableConnection(conf);
    return connection;
  }
```
Our call to `new BigtableConnection(conf)` leads to this chain:
`BigtableConnection.BigtableConnection():29` ->
`AbstractBigtableConnection.AbstractBigtableConnection():79` ->
`BigtableSession.BigtableSession():188` ->
`CredentialInterceptorCache.getCredentialsInterceptor():45` ->
`RefreshingOAuth2CredentialsInterceptor.asyncRefresh():165`

The line numbers refer to the .class files for these package versions:
com.google.cloud.bigtable.hbase1_x, 1.0.0-pre4
org.apache.hadoop.hbase.client, 1.0.0-pre4
com.google.cloud.bigtable.grpc, 1.0.0-pre4
com.google.cloud.bigtable.grpc.i, 1.0.0-pre4

In that `asyncRefresh()` method there's an external call which leads to a timeout when we don't have an internet connection. Is there a way I can avoid this?

**Comment by sduskis** [comment]
`conf.set("google.bigtable.auth.null.credential.enable", "true")` should work.

We should really have a simpler way to configure the emulator.

**Comment by stiancognite** [comment]
Thank you for replying!

It looks that I get the same thing even though I set that to `"true"`. I see that within `AbstractBigtableConnection`, there's a call to `BigtableOptionsFactory.fromConfiguration(conf)` where `conf` is read and a `BigtableOptions` is built. It doesn't look like `google.bigtable.auth.null.credential.enable` is read from the `conf` object there:
```
  public static BigtableOptions fromConfiguration(Configuration configuration) throws IOException {
    Builder bigtableOptionsBuilder = new Builder();
    bigtableOptionsBuilder.setProjectId(getValue(configuration, "google.bigtable.project.id", "Project ID"));
    bigtableOptionsBuilder.setInstanceId(getValue(configuration, "google.bigtable.instance.id", "Instance ID"));
    bigtableOptionsBuilder.setDataHost(getHost(configuration, "google.bigtable.endpoint.host", "bigtable.googleapis.com", "API Data"));
    bigtableOptionsBuilder.setTableAdminHost(getHost(configuration, "google.bigtable.admin.endpoint.host", "bigtableadmin.googleapis.com", "Table Admin"));
    bigtableOptionsBuilder.setInstanceAdminHost(getHost(configuration, "google.bigtable.instance.admin.endpoint.host", "bigtableadmin.googleapis.com", "Cluster Admin"));
    int port = configuration.getInt("google.bigtable.endpoint.port", 443);
    bigtableOptionsBuilder.setPort(port);
    bigtableOptionsBuilder.setUsePlaintextNegotiation(configuration.getBoolean("google.bigtable.use.plaintext.negotiation", false));
    setBulkOptions(configuration, bigtableOptionsBuilder);
    setChannelOptions(configuration, bigtableOptionsBuilder);
    setClientCallOptions(configuration, bigtableOptionsBuilder);
    return bigtableOptionsBuilder.build();
  }
```
The call to `new Builder()` there looks like this:
```
    public Builder() {
      this.dataHost = "bigtable.googleapis.com";
      this.tableAdminHost = "bigtableadmin.googleapis.com";
      this.instanceAdminHost = "bigtableadmin.googleapis.com";
      this.port = 443;
      this.dataChannelCount = BigtableOptions.BIGTABLE_DATA_CHANNEL_COUNT_DEFAULT;
      this.usePlaintextNegotiation = false;
      this.useCachedDataPool = false;
      this.retryOptions = (new com.google.cloud.bigtable.config.RetryOptions.Builder()).build();
      this.callOptionsConfig = (new com.google.cloud.bigtable.config.CallOptionsConfig.Builder()).build();
      this.credentialOptions = CredentialOptions.defaultCredentials();
    }
```
If I understand this correctly, I would like to have a call to `CredentialOptions.nullCredential()` at the last line there rather than a call to `CredentialOptions.defaultCredentials()`. Does that sound right?

**Comment by stiancognite** [comment]
Sorry, I missed your edit. It seems to work well when I set both of those. :) Thank you very much.

**Comment by sduskis** [comment]
I'm going to reopen this.  This should be a one line configuration.  This should be fixed with #1610.
